packages/@react-stately/data/test/useTreeData.test.js

```diff
@@ -753,8 +768,8 @@ describe('useTreeData', function () {
       result.current.moveBefore('David', ['John', 'Eli', 'Suzie']);
     });
     expect(result.current.items[0].key).toEqual('John');
-    expect(result.current.items[1].key).toEqual('Suzie');
-    expect(result.current.items[2].key).toEqual('Eli');
+    expect(result.current.items[1].key).toEqual('Eli');
+    expect(result.current.items[2].key).toEqual('Suzie');
```

Comment:
I'm surprised that this has `Eli` before `Suzie`, shouldn't `Suzie` be right after `John` since it was the child of `John`?

Reply:
ah never mind, I see that moveBefore will extract the child item and move it onto the same level as its parent so it specifically matters in what order the user selects the items.

I wonder is that expected? Or would a user expect that the selection order shouldn't matter? Should the parent child hierarchy be preserved?

Reply:
heh, I came here to comment the same thing. I think this ordering is unexpected. The order of the keys passed in to moveBefore shouldn't not affect the end order of items. I think we can think of the moveBefore as accepting a key and a Set, in which order doesn't matter nor is it guaranteed. It's a little incidental that we happen to pass an array, I think that was a mistake, but we can't undo it because then it wouldn't match out other hook.